test/abstractarray.jl

```diff
@@ -1280,3 +1280,33 @@ end
     @test Int[0 t...; t... 0] == [0 1 2; 1 2 0]
     @test_throws ArgumentError Int[t...; 3 4 5]
 end
+
+@testset "issue #39896, modified getindex " begin
+    #Testing types holds for the range types in Base that use 1-based indexing
+    a=collect(1:10)
+    ax=axes(a,1)
+    @test a==a[ax]
+    @test typeof(a)===typeof(a[ax])
+    a=reshape(collect(1:100),(10,10))
+    @test a[2:5]==2:5
+    a=collect(1:10)
+    ax=Base.OneTo(8)
+    @test a[1:8]==a[ax]
+    @test typeof(a[1:8])===typeof(a[ax])
+    a=reshape(collect(1:100),(10,10))
+    @test a[1:8,1]==a[ax,1]
+    @test typeof(a[1:8,1])===typeof(a[ax,1])
+    ax=UnitRange(1,10)
+    @test a[ax]==1:10
+    a=collect(1:BigInt(10))
```

Comment:
It seems wasteful to use `BigInt` here. Only the indexing range needs to be of type `UnitRange{BigInt}`. However, it might be good to test other eltypes, So maybe change it to `a = collect(1.0:10.0)`.